### PR TITLE
Csproj cleanup

### PR DESCRIPTION
**Pull request by mhutch**
Remove redundant information, and make more information redundant by setting good defaults in the MD props. This lays some groundwork for the PackageConfig migration and eventually SDK-style project format migration.

### CODE REVIEW

File: main/external/fsharpbinding/MonoDevelop.FSharp.Gui/MonoDevelop.FSharp.Gui.csproj [hunk 1]
```diff
@@ -10,10 +10,10 @@
     <OutputType>Library</OutputType>
     <AssemblyName>FSharpBinding.Gui</AssemblyName>
     <RootNamespace>MonoDevelop.FSharp.Gui</RootNamespace>
-    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
-    <TargetFrameworkProfile />
+    <TargetFrameworkVersion>$(MDFrameworkVersion)</TargetFrameworkVersion>
     <SignAssembly>true</SignAssembly>
     <AssemblyOriginatorKeyFile>..\..\..\msbuild\MonoDevelop-Public.snk</AssemblyOriginatorKeyFile>
+    <TargetFrameworkVersion>$(MDFrameworkVersion)</TargetFrameworkVersion>
```
Comment (mrward) [inline main/external/fsharpbinding/MonoDevelop.FSharp.Gui/MonoDevelop.FSharp.Gui.csproj]: TargetFrameworkVersion is defined twice here.

Comment (mhutch) [reply main/external/fsharpbinding/MonoDevelop.FSharp.Gui/MonoDevelop.FSharp.Gui.csproj]: @mrward fixed!

File: main/msbuild/MonoDevelop.BeforeCommon.targets [hunk 2]
```diff
@@ -4,26 +4,41 @@
...
 	</PropertyGroup>
 
-	<!-- force Roslyn compiler -->
-	<PropertyGroup Condition="'$(OS)'=='Unix' And '$(MSBuildProjectExtension)'=='.csproj'">
-		<_MonoBinDir>$(MSBuildBinPath)\..\..\..\4.5\</_MonoBinDir>
-		<CscToolExe>csc.exe</CscToolExe>
-		<CscToolPath>$(_MonoBinDir)</CscToolPath>
-		<DebugType>portable</DebugType>
-		<CscDebugFileExt>.pdb</CscDebugFileExt>
-		<_DebugFileExt>.pdb</_DebugFileExt>
```
Comment (nosami) [inline main/msbuild/MonoDevelop.BeforeCommon.targets]: We don't need this any longer? nice!

File: main/external/fsharpbinding/MonoDevelop.FSharpBinding/MonoDevelop.FSharp.fsproj [hunk 3]
```diff
@@ -49,13 +49,7 @@
     <GenerateXmlDoc>false</GenerateXmlDoc>
     <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
-  <PropertyGroup>
-    <FSharpTargetsPath>$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\FSharp\Microsoft.FSharp.Targets</FSharpTargetsPath>
-  </PropertyGroup>
-  <PropertyGroup Condition="'$(VisualStudioVersion)' == '10.0' OR '$(VisualStudioVersion)' == '11.0'">
```
Comment (nosami) [inline main/external/fsharpbinding/MonoDevelop.FSharpBinding/MonoDevelop.FSharp.fsproj]: This code was to support building inside Visual Studio on Windows but looks outdated now anyway.

**Comment by slluis** [comment]
The compatibility check step is failing, this should be fixed before merging.

**Pull request merged**